blockchain/blockchain.go

```diff
@@ -507,12 +534,16 @@ func (b *Blockchain) SanityCheckNewHeight(block *core.Block, stateUpdate *core.S
 }
 
 type txAndReceiptDBKey struct {
-	Number uint64
+	Number core.BlockNumber
 	Index  uint64
 }
 
 func (t *txAndReceiptDBKey) MarshalBinary() []byte {
-	return binary.BigEndian.AppendUint64(binary.BigEndian.AppendUint64([]byte{}, t.Number), t.Index)
+	numBytes, err := t.Number.MarshalBinary()
+	if err != nil {
+		panic(err)
```

Comment:
two comments:
- `BlockNumber.MarshalBinary()` always returns nil for an error (I added it because it's golang interface signature). So what do you guys think if I add `MustMarshalBinary(...) []byte` to `BlockNumber` ?
_ Do you think I should change return signature in `txAndReceiptDBKey.MarshalBinary()` also?

Reply:
>Do you think I should change return signature in txAndReceiptDBKey.MarshalBinary() also?

I think this